src/main/kotlin/com/expedia/graphql/generator/extensions/kParameterExtensions.kt

```diff
@@ -2,13 +2,17 @@ package com.expedia.graphql.generator.extensions
 
 import com.expedia.graphql.annotations.GraphQLContext
 import com.expedia.graphql.exceptions.CouldNotGetNameOfKParameterException
+import graphql.schema.DataFetchingEnvironment
 import kotlin.reflect.KParameter
 import kotlin.reflect.full.findAnnotation
+import kotlin.reflect.full.isSuperclassOf
 
 internal fun KParameter.isInterface() = this.type.getKClass().isInterface()
 
 internal fun KParameter.isGraphQLContext() = this.findAnnotation<GraphQLContext>() != null
 
+internal fun KParameter.isDataFetchingEnvironment() = this.type.getKClass().isSuperclassOf(DataFetchingEnvironment::class)
```

Comment:
Is this the correct direction? Shouldn't the parameter be the same or a subclass of `DataFetchingEnvironment`?

DFE is just an interface so we won't allow people to specify subclasses?

DFE also doesn't inherit anything so this check is basically is `this.type == DFE`